lib/pipeline.rb

```diff
@@ -1,32 +1,112 @@
 # frozen_string_literal: true
 
 require "rubygems"
-require "active_resource"
-require "pipeline/collection"
-require "pipeline/resource"
-require "pipeline/resources/account"
-require_relative "pipeline/resources"
-require_relative "pipeline/version"
-require_relative "pipeline/resources/definitions"
-
-Dir["#{File.dirname(__FILE__)}/resources/*.rb"].sort.each do |file|
-  p "requring #{file}"
-  require file
-end
 
-module Pipeline
-  class << self
+class Pipeline
+  VERSION = "1.0.0"
+
+  attr_reader :url, :prefix
+  attr_accessor :api_key, :app_key, :jwt
+
+  def initialize(url: "https://api.pipelinecrm.com", prefix: "/api/v3", api_key: nil, app_key: nil, jwt: nil)
+    @url = url
+    @prefix = prefix
+    @api_key = api_key
+    @app_key = app_key
+    @jwt = jwt
+  end
+
+  def authenticate(app_key:, email:, password:, mfa_code: nil)
+    @user = Pipeline::User.new(pipeline: self).authenticate(app_key: app_key, email: email, password: password, mfa_code: mfa_code)
+  end
+
+  def authenticated?
+    jwt.present? || (api_key.present? && app_key.present?)
+  end
+
+  def revoke_jwt
+    @user.revoke_jwt
+  end
 
-    def site
-      Pipeline::Resource.site
-    end
+  def renew_jwt
+    @user.renew_jwt
+  end
 
-    def site=(site)
-      Pipeline::Resource.site = site
-    end
+  def user
```

Comment:
The `user` and `account` methods are really ugly right now. They work, but not pretty.

Later, I'd like to implement a special method in the resource that will read from the specified endpoint and populate the attributes, instead of making two calls to the resource...

Reply:
Done, and added specs.